builtin/providers/aws/resource_aws_elasticsearch_domain.go

```diff
@@ -37,13 +37,9 @@ func resourceAwsElasticSearchDomain() *schema.Resource {
 				ForceNew: true,
 				ValidateFunc: func(v interface{}, k string) (ws []string, errors []error) {
 					value := v.(string)
-					if !regexp.MustCompile(`^[0-9A-Za-z]+`).MatchString(value) {
+					if !regexp.MustCompile(`^[a-z][0-9a-z\-]{2,27}$`).MatchString(value) {
```

Comment:
Those are the actual rules for es domain names, I had it failing in the API call locally if I didn't meet those.  Let me dig up the docs, one sec.